NuRadioReco/modules/channelThermalNoiseAdder.py

```diff
@@ -0,0 +1,282 @@
+import astropy.coordinates
+import functools
+import json
+import numpy as np
+import os
+import scipy.constants
+import scipy.interpolate
+from scipy.interpolate import interp1d
+from urllib.request import urlretrieve
+import warnings
+
+import NuRadioReco.detector.antennapattern
+import NuRadioReco.framework.channel
+import NuRadioReco.framework.sim_station
+from NuRadioReco.modules.base.module import register_run
+from NuRadioReco.utilities import units, ice, geometryUtilities
+
+import logging
+logger = logging.getLogger('NuRadioReco.channelThermalNoiseAdder')
+
+
+class channelThermalNoiseAdder:
+    """
+    Module to generate thermal noise from ice
+    This class is a stripped version of channelGalacticNoiseAdder
+    Intead of using a skymap, the module uses pre generated effective antenna temperatures,
+    generated by Steffen's effective temperature code which was modified by Felix
+    (see NuRadioMC/examples/simulate_effective_ice_temperature )
+    """
+
+    def __init__(self):
+        self.__n_side = None
+        self.__antenna_pattern_provider = NuRadioReco.detector.antennapattern.AntennaPatternProvider()
+
+
+    @functools.lru_cache(maxsize=1024 * 32)
+    def get_cached_antenna_response(self, antenna_pattern, zen, azi, *ant_orient):
+        return antenna_pattern.get_antenna_response_vectorized(self.freqs, zen, azi, *ant_orient)
+
+
+    def get_temperature_at_depth(self, depth):
+        grip_dir = f"{os.path.dirname(os.path.abspath(__file__))}/../../NuRadioMC/data"
+        grip_file = "griptemp.txt"
+        grip_path = grip_dir + "/" + grip_file
+
+        if not os.path.exists(grip_path):
+            grip_url = "https://doi.pangaea.de/10.1594/PANGAEA.89007?format=textfile"
+            urlretrieve(grip_url, grip_path)
+
+        grip_temp = np.loadtxt(grip_path, skiprows=39)
+        depth_values = -grip_temp[:,0]
+        temperature_values = grip_temp[:,1]
+        profile = interp1d(depth_values,
+                           temperature_values,
+                           bounds_error=False,
+                           fill_value=(temperature_values[-1],temperature_values[0]))
+        return profile(depth)
+
+    def solid_angle(self, theta, d_theta, d_phi):
+        return np.abs(np.sin(theta) * np.sin(d_theta / 2) * 2 * d_phi)
+
+    def get_temperature_from_json(self, temperature_file):
+        with open(temperature_file, "r") as file_open:
+            temperature_file_dict = json.load(file_open)
+        z_antenna = temperature_file_dict["z_antenna"]
+        theta = temperature_file_dict["theta"] * units.rad
+        eff_temperature = temperature_file_dict["eff_temperature"]
+        return z_antenna, theta, eff_temperature
+
+
+
+    def begin(self, sim_library_dir, nr_phi_bins=64, debug=False):
```

Comment:
The entire function is quite RNO-G specific. I am okay with keeping this module RNO-G specific but than we should move it the `/modules/RNOG` or remove the hard-coded things.